packages/workspace/src/utils/version-utils.ts

```diff
@@ -0,0 +1,21 @@
+import { clean } from 'semver';
+
+export function checkAndCleanWithSemver(version: string) {
+  let newVersion = version;
+
+  if (clean(newVersion)) {
+    return newVersion;
+  }
+
+  if (version.startsWith('~') || version.startsWith('^')) {
+    newVersion = version.substring(1);
+  }
+
+  if (!clean(newVersion)) {
```

Comment:
You're using `clean` but not actually returning the cleaned result. Did you mean to use `valid` instead?